### DISCUSSION TITLE
Having trouble with duplicate variables being output

### DISCUSSION THREAD

**Discussion by dovrosenberg** [primefaces/community#2968] [PrimeVue]
I'm getting started with PrimeVue and seeing a strange behavior.  It's outputting multiple copies of variables that I am overriding in a preset - one for the default and one for my override.  The order is random, so sometimes it uses mine and sometimes it uses the default, depending on which gets output second.

This is my preset definition:
```
import Aura from '@primevue/themes/aura';
import { definePreset } from '@primevue/themes';

export default definePreset(Aura, {
  components: {
    datatable: {
      headerBackground: 'red',
      headerCellBackground: 'red',
      rowBackground: 'red',
      rowStripedBackground: 'red',
    },
  }
});
```

The output `style` block includes (for example) both `p-datatable-header-background:red` and `p-datatable-header-background:var(--p-content-background)`

I also tried overriding `--p-content-background` and then I got two definitions of that variable.

Am I doing something wrong? I assume this isn't the normal behavior.

Thanks!

**Comment by j0rgedev** [comment]
Hello @dovrosenberg.  I tried to reproduce what you said but I don't get to understand what you mean. This is the output style I get. What's wrong with it? 🤔
![image](https://github.com/user-attachments/assets/eaa0b8e6-8261-42c5-be6f-6863ae282d1c)

**Comment by dovrosenberg** [comment]
Yes - sometimes it works - it depends on the specific variable and the preset (changing from Lara to Aura changes the behavior for me).  So I guessed I picked a bad example.  If you change to Lara, I think that particular one will no longer work.  More broadly, if you search in the debugger Elements tab for `--p-datatable-header-background` you will (or at least I do) see two different definitions of the variable.  Depending on whether you use Lara or Aura, the order may be flipped.  Whichever comes second is the color you end up getting.  

Actually, when I did it this time to grab a screenshot, I seem to have 3 definitions of the variable... I hadn't noticed that before - previously I've only noticed 2.  But same issue.
![image](https://github.com/user-attachments/assets/2bec8df8-f266-482b-bea0-2106db67c896)

**Reply by j0rgedev** [reply]
Hello @dovrosenberg, apologies for the delay. I was able to replicate the issue, and it appears this may indeed be a potential problem. Here’s why: Aura and Lara themes have different theming tokens, which might not be expected.

In the [Aura DataTable Theme](https://github.com/primefaces/primevue/blob/master/packages/themes/src/presets/aura/datatable/index.js#L5), there is a header background color token that you can later override when creating your custom preset. However, in the [Lara DataTable Theme](https://github.com/primefaces/primevue/blob/master/packages/themes/src/presets/lara/datatable/index.js#L5), the header background color token is missing. Because of this inconsistency, you’re likely seeing two different definitions of that variable. 

This discrepancy might be the root cause of this 'issue' and perhaps someone from the team could give us more insight or guidance  👀

**Comment by dovrosenberg** [comment]
Thank you so much for following up.  I've seen this with other items too, for example [`button.primary.hover.background` ](https://primevue.org/button/), but now that you've shown me where to look, I can see that the actual declaration doesn't quite match the documentation (it's actually `button.colorScheme.light.primary.hover.background`).

Really appreciate the help, and I'll leave this open in case the Lara DataTable one is a bug.
